tests/unit/mixins/indexProvider.spec.js

```diff
@@ -0,0 +1,521 @@
+        },
+        {
+          id: 'Child0_GrandChild1',
+          title: 'Child0_GrandChild1',
+          path: '/documentation/foo/child0/grandchild1',
+          type: 'tutorial',
+          children: [{
+            id: 'Child0_GrandChild0_GreatGrandChild0',
+            title: 'Child0_GrandChild0_GreatGrandChild0',
+            path: '/documentation/foo/child0/grandchild0/greatgrandchild0',
+            type: 'tutorial',
+          }],
+        },
+        {
+          id: 'Child0_GrandChild2',
+          title: 'Child0_GrandChild2',
+          path: '/documentation/foo/child0/grandchild2',
+          type: 'tutorial',
+        },
+      ],
+    },
+    {
+      id: 'Child1',
+      title: 'Child1',
+      path: '/documentation/foo/child1/',
+      type: 'tutorial',
+      children: [{
+        id: 'Child1_GrandChild0',
+        title: 'Child1_GrandChild0',
+        path: '/documentation/foo/child1/grandchild0',
+        type: 'method',
+      }],
+    },
+  ],
+};
+
+const flatChildren = {
+  [Language.swift.key.url]: [
+    {
+      uid: -196255993,
+      parent: '<root>',
+      index: 0,
+      siblingsCount: 3,
+      depth: 0,
+      childUIDs: [],
+    },
+    {
+      uid: -196255992,
+      parent: '<root>',
+      index: 1,
+      siblingsCount: 3,
+      depth: 0,
+      childUIDs: [],
+    },
+    {
+      uid: -196255991,
+      parent: '<root>',
+      index: 2,
+      siblingsCount: 3,
+      depth: 0,
+      childUIDs: [],
+    },
+  ],
+  [Language.objectiveC.key.url]: [
+    {
+      uid: -196255993,
+      parent: '<root>',
+      index: 0,
+      siblingsCount: 1,
+      depth: 0,
+      childUIDs: [],
+    },
+  ],
+};
+
+const swiftIndexOne = {
+  id: 'foo',
+  path: technologyUrl,
+  children: [1, 2, 3],
+};
+const objectiveCIndexOne = {
+  id: 'foo-objc',
+  path: technologyUrl,
+  children: [1],
+};
+
+const references = {
+  foo: { bar: 'bar' },
+};
+
+const includedArchiveIdentifiers = [
+  'foo',
+  'bar',
+];
+
+const interfaceLanguages = {
+  [Language.swift.key.url]: [
+    swiftIndexOne,
+  ],
+  [Language.objectiveC.key.url]: [
+    objectiveCIndexOne,
+  ],
+};
+
+const response = {
+  includedArchiveIdentifiers,
+  interfaceLanguages,
+  references,
+};
+
+const technologyProps = {
+  [Language.swift.key.url]: {
+    technology: undefined,
+    technologyPath: technologyUrl,
+    isTechnologyBeta: undefined,
+  },
+  [Language.objectiveC.key.url]: {
+    technology: undefined,
+    technologyPath: technologyUrl,
+    isTechnologyBeta: undefined,
+  },
+};
+
+fetchData.mockReturnValue(response);
+
+const Component = {
+  name: 'MyComponent',
+  template: '<div/>',
+  mixins: [indexProvider],
+};
+
+const createWrapper = ({ mocks } = {}) => shallowMount(Component, { mocks });
+
+describe('indexProvider', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  afterEach(() => {
+    IndexStore.reset();
+  });
+
+  it('fetches data when parent component imports `indexProvider`', async () => {
```

Comment:
It fetches data when the path for the index data changes and not when the mixin is used, right?